geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/expression/FastPropertyName.scala

```diff
@@ -8,47 +8,22 @@
 
 package org.locationtech.geomesa.filter.expression
 
-import com.typesafe.scalalogging.LazyLogging
+import org.geotools.filter.expression.PropertyAccessor
 import org.geotools.util.Converters
-import org.locationtech.geomesa.utils.geotools.{SimpleFeaturePropertyAccessor, SimpleFeatureTypes}
 import org.opengis.feature.simple.SimpleFeature
-import org.opengis.filter.expression.{ExpressionVisitor, PropertyName}
+import org.opengis.filter.expression.{Expression, ExpressionVisitor, PropertyName}
 import org.xml.sax.helpers.NamespaceSupport
 
 /**
  * Implementation of property name that looks up the value by index
  */
-class FastPropertyName(name: String) extends PropertyName with org.opengis.filter.expression.Expression
-  with SimpleFeaturePropertyAccessor with LazyLogging {
-
-  private var getProperty: (SimpleFeature) => AnyRef = null
+class FastPropertyName(name: String, index: Int) extends PropertyName with Expression {
 
   override def getPropertyName: String = name
 
   override def getNamespaceContext: NamespaceSupport = null
 
-  override def evaluate(obj: AnyRef): AnyRef = {
-    val sf = try {
-      obj.asInstanceOf[SimpleFeature]
-    } catch {
-      case e: Exception => throw new IllegalArgumentException("Only simple features are supported", e)
-    }
-    if (getProperty == null) {
-      val nsIndex = name.indexOf(':')
-      val localName = if (nsIndex == -1) name else name.substring(nsIndex + 1)
```

Comment:
it went into the pre-processing in FastFilterFactory: https://github.com/elahrvivaz/geomesa/blob/f70e70b837cf3fd72504b4558ba2f0cd03a1a9be/geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/factory/FastFilterFactory.scala#L32